megamek/src/megamek/client/ui/swing/tooltip/SBFFormationTooltip.java

```diff
@@ -0,0 +1,208 @@
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.swing.tooltip;
+
+import megamek.client.ui.swing.GUIPreferences;
+import megamek.client.ui.swing.util.UIUtil;
+import megamek.common.*;
+import megamek.common.annotations.Nullable;
+import megamek.common.strategicBattleSystems.SBFFormation;
+import megamek.common.strategicBattleSystems.SBFUnit;
+
+import java.util.List;
+import java.awt.*;
+import java.text.MessageFormat;
+import java.util.Set;
+
+public final class SBFFormationTooltip {
+
+    private static final GUIPreferences GUIP = GUIPreferences.getInstance();
+    private static final String SHORT = "&nbsp;";
+    private static final Set<String> ABBREV_NAME_PARTS_UNIT = Set.of("Lance", "Squadron", "Wing", "Flight");
+
+    public static String getTooltip(List<SBFFormation> formations, @Nullable IGame game) {
+        StringBuilder result = new StringBuilder("<HTML><HEAD>");
+        result.append(styles());
+        result.append("</HEAD><BODY>");
+        for (SBFFormation formation : formations) {
+            Player owner = (game != null) ? game.getPlayer(formation.getOwnerId()) : null;
+            Color ownerColor = (owner != null) ? owner.getColour().getColour() : Color.BLACK;
+            String styleColor = Integer.toHexString(ownerColor.getRGB() & 0xFFFFFF);
+            result.append("<div style=\"padding:0 10; border:2; margin: 5 0; border-style:solid; border-color:" + styleColor + ";\">");
+            result.append(nameLines(formation, game));
+            result.append(formationStats(formation));
+            result.append("</div>");
+        }
+        result.append("</BODY></HTML>");
+        return result.toString();
+    }
+
+    public static String getTooltip(SBFFormation formation, @Nullable IGame game) {
+        StringBuilder result = new StringBuilder("<HTML><HEAD>");
+        result.append(styles());
+        result.append("</HEAD><BODY>");
+        Player owner = (game != null) ? game.getPlayer(formation.getOwnerId()) : null;
+        Color ownerColor = (owner != null) ? owner.getColour().getColour() : Color.BLACK;
+        String styleColor = Integer.toHexString(ownerColor.getRGB() & 0xFFFFFF);
+        result.append("<div style=\"padding:0 10; border:2; border-style:solid; border-color:" + styleColor + ";\">");
+        result.append(nameLines(formation, game));
+        result.append(formationStats(formation));
+        result.append("</div>");
+        result.append("</BODY></HTML>");
+        return result.toString();
+    }
+
+    private static String asCSS(String cssClass, String content) {
```

Comment:
should these be an a common html utils class?

Reply:
Actually that class was no longer used, but true. Will change.

Reply:
Yes possibly. Will have a look